stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/Write.java

```diff
@@ -57,32 +62,67 @@
      */
     private static final int CONTENT_INDEX = 0;
 
-    /**
+    /*
      * Index which holds the start offset in ballerina.io#writeBytes.
      */
     private static final int START_OFFSET_INDEX = 0;
 
+    /*
+     * Index which holds the number of bytes which should be written.
+     */
+    private static final int NUMBER_OF_BYTES_INDEX = 1;
+
+    private static final Logger log = LoggerFactory.getLogger(Write.class);
+
+    /*
+     * Function which will be notified on the response obtained after the async operation.
+     *
+     * @param result context of the callback.
+     * @return Once the callback is processed we further return back the result.
+     */
+    /*private static EventResult writeResponse(EventResult<Integer, EventContext> result) {
+        *//*
+         * Async task response goes here
+         *//*
+        EventContext eventContext = result.getContext();
+        Context context = eventContext.getContext();
+        Throwable error = eventContext.getError();
+        Integer numberOfBytesWritten = result.getResponse();
+        BStruct errorStruct;
+        if (null != error) {
+            errorStruct = IOUtils.createError(context, error.getMessage());
+        }
+        return result;
+    }*/
+
+
     /**
      * Writes bytes to a given channel.
-     *
+     * <p>
      * {@inheritDoc}
      */
     @Override
     public BValue[] execute(Context context) {
         BStruct channel;
         byte[] content;
-        int startOffset;
-        int numberOfBytesWritten;
+        int numberOfBytes;
+        int offset;
```

Comment:
Shall we define above 4 variables (channel, content, numberOfBytes, offset) within the try block it self. That will reduce them to a single line and more readable

Reply:
In terms of readability, having variables defined outside the try block would help understand it's usage, and the types of memory allocated for the whole process. And some of the variables are being used by catch blocks etc which will not allow to move them under the try scope